kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/analyze/TopicAnalyzeService.java

```diff
@@ -0,0 +1,158 @@
+package com.provectus.kafka.ui.service.analyze;
+
+import com.provectus.kafka.ui.model.KafkaCluster;
+import com.provectus.kafka.ui.model.TopicAnalyzeStateDTO;
+import com.provectus.kafka.ui.service.ConsumerGroupService;
+import com.provectus.kafka.ui.service.TopicsService;
+import com.provectus.kafka.ui.util.OffsetsSeek.WaitingOffsets;
+import java.io.Closeable;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.errors.InterruptException;
+import org.apache.kafka.common.errors.WakeupException;
+import org.apache.kafka.common.utils.Bytes;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class TopicAnalyzeService {
+
+  private final AnalyzeTasksStore analyzeTasksStore = new AnalyzeTasksStore();
+
+  private final TopicsService topicsService;
+  private final ConsumerGroupService consumerGroupService;
+
+  public Mono<Void> analyze(KafkaCluster cluster, String topicName) {
+    return topicsService.getTopicDetails(cluster, topicName)
+        .doOnNext(topic ->
+            startAnalyze(
```

Comment:
If the analysis is already started for the topic we'll return this
`throw new IllegalStateException("Topic is already analyzing");`
I hope frontend could get some well-written error message instead.